polars/polars-lazy/polars-pipe/src/pipeline/convert.rs

```diff
@@ -224,89 +224,91 @@ where
-                            groupby_out_schema.with_column(name.clone(), dtype.clone());
-                            expr_arena.add(AExpr::Column(Arc::from(key.as_str())))
-                        })
-                        .collect();
-
-                    let aggs = input_schema
-                        .iter_names()
-                        .flat_map(|name| {
-                            if key_names.contains(name.as_str()) {
-                                None
-                            } else {
-                                let (_, name, dtype) =
-                                    input_schema.get_full(name.as_str()).unwrap();
-                                groupby_out_schema.with_column(name.clone(), dtype.clone());
-                                let col = expr_arena.add(AExpr::Column(Arc::from(name.as_str())));
-                                Some(match options.keep_strategy {
-                                    UniqueKeepStrategy::First | UniqueKeepStrategy::Any => {
-                                        expr_arena.add(AExpr::Agg(AAggExpr::First(col)))
-                                    }
-                                    UniqueKeepStrategy::Last => {
-                                        expr_arena.add(AExpr::Agg(AAggExpr::Last(col)))
-                                    }
-                                    UniqueKeepStrategy::None => {
-                                        unreachable!()
-                                    }
-                                })
-                            }
-                        })
-                        .collect();
-                    (keys, aggs, groupby_out_schema.into())
-                }
-            };
-
-            let key_columns = Arc::new(exprs_to_physical(
-                &keys,
-                expr_arena,
-                to_physical,
-                Some(&input_schema),
-            )?);
-
-            let mut aggregation_columns = Vec::with_capacity(aggs.len());
-            let mut agg_fns = Vec::with_capacity(aggs.len());
-            let mut input_agg_dtypes = Vec::with_capacity(aggs.len());
-
-            for node in &aggs {
-                let (input_dtype, index, agg_fn) =
-                    convert_to_hash_agg(*node, expr_arena, &input_schema, &to_physical);
-                aggregation_columns.push(index);
-                agg_fns.push(agg_fn);
-                input_agg_dtypes.push(input_dtype);
-            }
-            let aggregation_columns = Arc::new(aggregation_columns);
-
-            let groupby_sink = Box::new(GenericGroupby2::new(
-                key_columns,
-                aggregation_columns,
-                Arc::from(agg_fns),
-                output_schema,
-                input_agg_dtypes,
-                options.slice,
-            ));
-
-            Box::new(ReProjectSink::new(input_schema, groupby_sink))
+        Unique { input, subset, options } => {
+            todo!()
```

Comment:
@ritchie46 I think I have most everything figured out except for this function. I haven't done much with the OOC planner yet, so I'm not really sure how this should be implemented. Is there anywhere else that i could copy from?